### PR TITLE
Display offers

### PR DESCRIPTION
**Pull request by starjardin**
Hi team,

Display offers task has been implemented. Please have a look at it. Comments are highly appreciated.
Thanks.
Haukea.

### CODE REVIEW

**Review by ravinalamada — APPROVED** [review]

**Review by Fodilahy-mena — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -1,34 +1,23 @@
...
-      <Route exact path='/transactions' component={Transactions} />
-      <Route exact path='/signup' component={RoleSelectOnboard} />
-      <Route component={NotFound} />
-    </Switch>
-  )
+	return (
+		<Switch>
+			<Route exact path='/' component={OnboardingPage} />
+			<Route exact path='/home' component={HomePage} />
+			<Route exact path={['/signin', '/log in', '/login']} component={SignIn} />
```
Comment (Fodilahy-mena) [inline src/App.tsx:15]: We should be more consistent about this path but I think it's fine for now. Great approach ha!

File: src/Pages/HomePage.tsx [hunk 2]
```diff
@@ -1,188 +1,112 @@
-import styled from "styled-components";
-import { useEffect } from "react";
-import { mediaQueriesPx } from "../mediaQueries";
+import styled from 'styled-components'
```
Comment (andrzej-onja) [inline src/Pages/HomePage.tsx:1]: Please merge Home into HomePage

File: src/Pages/HomePage.tsx [hunk 3]
```diff
@@ -1,188 +1,112 @@
...
+					<PendingIndicator alt='PendingSmallIcon' size='s' />
+				) : (
+					<CategoryList
+						categories={categories}
+						primary={true}
+						selectCategory={() => {}}
+					/>
+				)}
+				<Separator />
+				<OffersPage />
```
Comment (andrzej-onja) [inline src/Pages/HomePage.tsx:48]: OffersPage is not a page it is a component that should accept offer list as a prop

File: src/Pages/OffersPage.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
+import { fetchOffers } from '../redux/slices/offersSlice'
```
Comment (andrzej-onja) [inline src/Pages/OffersPage.tsx:1]: please see comment above this should be a list of offers component. Prop that accepts list of offers and state. Fetching logic is done in parent

File: src/components/Back/Back.tsx [hunk 5]
```diff
@@ -3,63 +3,58 @@ import styled from 'styled-components'
...
+		height='16'
+		viewBox='0 0 10 16'
+		fill='none'
+		xmlns='http://www.w3.org/2000/svg'>
+		<path
+			d='M9.37604 1.90245L8.04104 0.574951L0.623535 7.99995L8.04854 15.4249L9.37604 14.0975L3.27854 7.99995L9.37604 1.90245Z'
+			fill='currentColor'
+		/>
+	</svg>
 )
```
Comment (andrzej-onja) [inline src/components/Back/Back.tsx:17]: lets move it to a separate standalone file like other svg above

File: src/components/Offer/Offer.tsx [hunk 6]
```diff
@@ -7,185 +7,202 @@ import { ImageWithinOffer } from '../ImageWithinOffer/ImageWithinOffer'
...
+	ratingDescription,
+	amountOfProduct,
+	offerDescription,
+	isLearnEnabled,
+	detailButtonText,
+	favoriteButtonText,
+	isFavourited,
+	profile,
+	name,
+	location,
```
Comment (andrzej-onja) [inline src/components/Offer/Offer.tsx:55]: I would prefer if offer would be an object with defined interface that is also used in offerSlice. This way we avoid retyping each prop

File: src/components/Offer/Offer.tsx [hunk 7]
```diff
@@ -7,185 +7,202 @@ import { ImageWithinOffer } from '../ImageWithinOffer/ImageWithinOffer'
...
+				fill='#041D42'
+			/>
+		</svg>
```
Comment (andrzej-onja) [inline src/components/Offer/Offer.tsx:70]: move svg into a file

File: src/graphqlHelper.ts [hunk 8]
```diff
@@ -1,19 +1,19 @@
...
-};
+	return axios.post('http://localhost:4000/graphql?', {
+		query,
+	})
+}
 
 export const getAdsQuery = () => {
-  return `{
+	return `{
           ads{title, createdAt, id, username,comments{body, id, username}}
```
Comment (andrzej-onja) [inline src/graphqlHelper.ts:11]: add `user{firstName, lastName etc}` this way you can get all information required to display offer

File: src/redux/slices/adsSlice.ts [hunk 9]
```diff
@@ -1,106 +1,106 @@
...
-  message: string
+	message: string
 }
 
 // possible errors.
 export const fetchAds = createAsyncThunk<
-  any[],
-  number,
-  { rejectValue: FetchAdsError }
+	any[],
```
Comment (andrzej-onja) [inline src/redux/slices/adsSlice.ts:16]: As mentioned above define offer interface and replace any

File: src/redux/slices/adsSlice.ts [hunk 10]
```diff
@@ -1,106 +1,106 @@
...
+		decrement: (state) => {
+			// console.log(state.value, 'dec')
+			state.value -= 1
+		},
+		//TODO remove used only as an example
+		incrementByAmount: (state, action: PayloadAction<number>) => {
+			// console.log(action, 'incbyamAC')
+			// console.log(state, 'increment')
+			state.value += action.payload
+		},
```
Comment (andrzej-onja) [inline src/redux/slices/adsSlice.ts:71]: Its time to  remove all of the above, those were used as an example

File: src/redux/slices/adsSlice.ts [hunk 11]
```diff
@@ -1,106 +1,106 @@
...
-    })
-  },
+		// When a server responses with an error:
+		builder.addCase(fetchAds.rejected, (state, { payload }) => {
+			// We show the error message
+			// and change `status` back to `idle` again.
+			// console.log(payload, 'REjected error')
+			if (payload) state.error = payload
+			state.status = 'idle'
+		})
```
Comment (andrzej-onja) [inline src/redux/slices/adsSlice.ts:102]: remove all the comments we dont need them now

File: src/redux/slices/offersSlice.ts [hunk 12]
```diff
@@ -0,0 +1,58 @@
+import { createAsyncThunk, createSelector, createSlice } from '@reduxjs/toolkit'
```
Comment (andrzej-onja) [inline src/redux/slices/offersSlice.ts:1]: If you are adding offerSlice,  adsSlice can be removed - its the same thing

File: src/redux/store.ts [hunk 13]
```diff
@@ -1,23 +1,25 @@
...
-export const history = createBrowserHistory();
+export const history = createBrowserHistory()
 
 const rootReducer = combineReducers({
-  router: connectRouter(history) as any,
-  ads: adsReducer,
-  categories: categoriesReducer
-});
+	router: connectRouter(history) as any,
+	ads: adsReducer,
```
Comment (andrzej-onja) [inline src/redux/store.ts:13]: As comment above ads and offers is the same thing. Remove one of them

**Pull request closed without merging**